.github/workflows/test-measure.yml

```diff
@@ -0,0 +1,289 @@
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Configure Composer cache
+        uses: actions/cache@v3.0.2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --optimize-autoloader --no-progress --no-interaction --no-scripts
+
+      - name: Validate composer.json
+        run: composer --no-interaction validate --no-check-all
+
+      - name: Detect coding standard violations (PHPCS)
+        run: vendor/bin/phpcs -q --report=checkstyle --runtime-set ignore_errors_on_exit 1 --runtime-set ignore_warnings_on_exit 1 | cs2pr --graceful-warnings
+
+  unit-test-php:
+    name: "PHP Unit test"
+    runs-on: ubuntu-latest
+    needs: pre-run
+    steps:
+      # Note: The repeated `needs.pre-run.outputs.changed-php-count > 0` checks would be avoided if a step could short-
+      # circuit per <https://github.com/actions/runner/issues/662>. The reason why the if statement can't be put on the
+      # job as a whole is because the name is variable based on the matrix, and if the condition is not met then the
+      # name won't be interpolated in order to match the required jobs set up in branch protection.
+      - name: Notice
+        if: needs.pre-run.outputs.changed-php-count > 0
+        run: echo "No PHP files were changed in CK theme so no PHP unit tests will run"
+
+      - name: Checkout
+        if: needs.pre-run.outputs.changed-php-count > 0
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        if: needs.pre-run.outputs.changed-php-count > 0
+        uses: actions/setup-node@v3.2.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+
+      - name: Install Node dependencies
+        if: needs.pre-run.outputs.changed-php-count > 0
+        run: npm ci --ignore-scripts
+        env:
+          CI: true
+
+      - name: Start WP environment
+        if: needs.pre-run.outputs.changed-php-count > 0
+        run: npm run wp-env start
+
+      - name: Run tests
+        if: ${{ needs.pre-run.outputs.changed-php-count > 0 }}
+        run: npm run test:php
+
+  build-prod:
+    name: 'Build production'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3.2.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+
+      - name: Install Node dependencies
+        run: npm ci --ignore-scripts
+        env:
+          CI: true
+
+      - name: Build production
+        id: build
+        run: npm run build:prod
+        continue-on-error: true
+        env:
+          CI: true
+
+      - name: Comment on PR with build status
```

Comment:
Do we need to add these labels for whether or not build is passing on a PR?

Reply:
Not mandatory but it can be better for visibility if the build is passing for a PR. 

Actually, it should be added only if CSS or JS file is edited. What you say.

Reply:
I don't find labeling for builds to be necessary; even for JS & CSS files. But don't have any points to oppose it either as it's just my preference, so we can go ahead.